### DISCUSSION TITLE
What's the advantage of LaunchUriAsync over Process.Start to launch default browser with uri?

### DISCUSSION THREAD

**Discussion by TangDave** [AvaloniaUI/Avalonia#17543] [Q&A] [answered]
On Windows, Process.Start has always launched the default browser just fine, but have run into some problems with Linux - is LaunchUriAsync designed to work on Linux more reliably? Is there a reason to prefer LaunchUriAsync even for Windows?

**Comment by maxkatz6** [comment, marked as answer]
LaunchUriAsync is designed to abstract launching URI code from platform specific details. Even `Process.Start` doesn't have a specific right way to do that for all desktop platform, you need to write slightly different commands for different OS. And obviously it won't work for non-desktop platforms.

> but have run into some problems with Linux - is LaunchUriAsync designed to work on Linux more reliably?

It should work. Not necessary more reliably, but definitely easier to use.
If there is a problem, you can report an issue.
